src/main/frontend/src/components/aggregates/AggregateDialog.svelte

```diff
@@ -18,94 +12,83 @@
-		rootPath = "/";
-		routes = [];
-		producerExchangeName = "";
-		schemaGroup = retrieveSchemaGroup();
-		disableSchemaGroup = !canWriteSchemaGroup();
-		outgoingEvents = [];
-		consumerExchangeName = "";
-		receivers = [];
+	const add = () => {
+		if(!valid) return;
+		$aggregateSettings = [...$aggregateSettings, newAggregate];
+		closeDialog();
 	}
 
-	const retrieveSchemaGroup = () => {
-		return $aggregateSettings.length > 0 ? $aggregateSettings[0].producerExchange.schemaGroup : "";
+	const update = () => {
+		if(!valid) return;
+		$aggregateSettings = $aggregateSettings.filter(a => JSON.stringify(a) !== JSON.stringify(oldAggregate));
+		$aggregateSettings = [...$aggregateSettings, newAggregate];
+		closeDialog();
 	}
 
-	const canWriteSchemaGroup = () => {
-		return currentId == 0 || (schemaGroup == undefined && schemaGroup.length == 0);
+	const closeDialog = () => {
+		clearFields();
+		dialogActive = false;
+		setLocalStorage("aggregateDialogState", {});
 	}
 
-	$: changedCurrent(currentId);
-	function changedCurrent(index) {
-		aggregateIndex = index;
-		if(index !== undefined && $aggregateSettings[index]) {
-			const aggregateWithId =  $aggregateSettings[index];
-			aggregateName = aggregateWithId.aggregateName;
-			stateFields = aggregateWithId.stateFields;
-			events = aggregateWithId.events;
-			methods = aggregateWithId.methods;
-			rootPath = aggregateWithId.api.rootPath;
-			routes = aggregateWithId.api.routes;
-			producerExchangeName = aggregateWithId.producerExchange.exchangeName;
-			schemaGroup = aggregateWithId.producerExchange.schemaGroup;
-			outgoingEvents = aggregateWithId.producerExchange.outgoingEvents;
-			consumerExchangeName = aggregateWithId.consumerExchange.exchangeName;
-			receivers = aggregateWithId.consumerExchange.receivers;
-		} else {
-			reset();
-		}
+	const initFieldsWith = (aggregate) => {
+		aggregateName = aggregate.aggregateName;
+		stateFields = aggregate.stateFields;
+		events = aggregate.events;
+		methods = aggregate.methods;
+		rootPath = aggregate.api.rootPath;
+		routes = aggregate.api.routes;
+		producerExchangeName = aggregate.producerExchange.exchangeName;
+		schemaGroup = aggregate.producerExchange.schemaGroup;
+		outgoingEvents = aggregate.producerExchange.outgoingEvents;
+		consumerExchangeName = aggregate.consumerExchange.exchangeName;
+		receivers = aggregate.consumerExchange.receivers;
+		disableSchemaGroup = aggregate.disableSchemaGroup;
 	}
+	const clearFields = () => initFieldsWith(initialAggregate);
 
-	const validField = (f) => !identifierRule(f.name) && f.type && !isPropertyUnique(f.name, stateFields, 'name');
-	const validEvent = (e) => !classNameRule(e.name) && e.fields.length > 0 && !isPropertyUnique(e.name, events, 'name');
-	const validMethod = (m) => !identifierRule(m.name) && m.parameters.length > 0 && m.event && !isPropertyUnique(m.name, methods, 'name');
+	const validField = (f) => !identifierRule(f.name) && f.type && !isPropertyUniqueRule(f.name, stateFields, 'name');
```

Comment:
I renamed to "..Rule", otherwise it looks like it's valid if the property is _not_ unique, now it's slightly better